peak/peak.py

```diff
@@ -66,7 +72,41 @@ def call_wrapper(*args,**kwargs):
         return call_wrapper
     return decorator
 
+#This decorator does the following:
+#1) Caches the function call
+#2) Stores the family closure in Peak._fc_
+def family_closure(f):
+    num_inputs = f.__code__.co_argcount
+    if num_inputs != 1:
+        raise ValueError("Family Closure must take a single input 'family'")
+    @functools.lru_cache(None)
+    @functools.wraps(f)
+    def fc(family):
+        cls = f(family)
+        if not (isclass(cls) and issubclass(cls,Peak)):
+            raise ValueError("Family closure must return a peak class")
```

Comment:
why?

Reply:
Not sure what you mean. I want to guarantee that the return value is a single peak class.

Reply:
Why do you want that?  Why is it necessary to inherit peak.  If everything works what does it matter if it inherits from peak?

Reply:
Inheriting from Peak implies I can get the input_t and output_t. Mapper does not work on non-Peak classes.

Reply:
You still haven't told me any reason why we shouldn't use duck typing.  

If the mapper only works on classes whose call method have an input_t and output_t that is fine. Throw an error there.  Even mention that they should inherit from Peak.  However, if I manually build a class that looks like a Peak class that should be fine.  Unless there is some reason why it isn't.

Reply:
Depends on your philosophy on building am embedded DSL. I am pro forcing users to adhere to a particular style that allows the DSL to work and to eagerly inform them with userful error messages when they inevitably screw up. This seems to be more important than allowing power users to unsafely ducktype. If you fundamentally disagree with this, then I'll change it. Maybe @phanrahan can comment.

Reply:
I'd be fine with a warning.  But errors should be errors in my mind.

Reply:
Changed to Warning